File: src/main/java/ru/ioleksiv/telegram/bot/api/annotations/filter/secondary/PassportFileArrayFilter.java
package ru.ioleksiv.telegram.bot.api.annotations.filter.secondary;

import ru.ioleksiv.telegram.bot.api.model.annotation.AnnotationState;

import java.lang.annotation.Documented;
import java.lang.annotation.ElementType;
import java.lang.annotation.Retention;
import java.lang.annotation.RetentionPolicy;
import java.lang.annotation.Target;

@Retention(RetentionPolicy.RUNTIME)
@Target(ElementType.METHOD)
@Documented
public @interface PassportFileArrayFilter {

    AnnotationState value() default AnnotationState.ON;

    PassportFileFilter[] array() default @PassportFileFilter(AnnotationState.OFF);

}

File: core/src/main/kotlin/io/github/oleksivio/tl/kbot/core/annotations/filter/composite/PassportFileArrayFilter.kt
package io.github.oleksivio.tl.kbot.core.annotations.filter.composite

import io.github.oleksivio.tl.kbot.core.annotations.filter.primitive.IntegerFilter
import io.github.oleksivio.tl.kbot.core.model.annotation.AnnotationState

@Retention
@Target(AnnotationTarget.FUNCTION, AnnotationTarget.PROPERTY_GETTER, AnnotationTarget.PROPERTY_SETTER)
@MustBeDocumented
annotation class PassportFileArrayFilter(
    val status: AnnotationState = AnnotationState.ON,
    val validator: Array<String> = [],
    val fileSize: IntegerFilter = IntegerFilter(
        status = AnnotationState.OFF
    ),
    val fileDate: IntegerFilter = IntegerFilter(
        status = AnnotationState.OFF
    )
)
